superset/assets/src/explore/components/controls/SubscriberLayer.jsx

```diff
@@ -0,0 +1,378 @@
+
+    this.setState({
+      operatorType,
+      subscriptionList,
+      subscribe_columns,
+      allowMoreColumns: subscribe_columns[index]['col'] && subscribe_columns[index]['op'] ? true : false,
+    });
+  }
+
+  addMoreColumns() {
+    const subscriptionList = this.state.subscriptionList;
+    const data = subscriptionList[subscriptionList.length - 1];
+    const newData = { columnType: '', operatorType: '', index: data.index + 1 }
+    this.setState(prevState => ({ subscriptionList: [...prevState.subscriptionList, newData] }))
+  }
+
+  removeColumn(e, column) {
+    this.setState(prevState => ({ subscriptionList: [...prevState.subscriptionList.filter(x => x.index !== column.index)] }))
+  }
+
+  deleteSubscriber() {
+    this.props.close();
+    if (!this.state.isNew) {
+      this.props.removeSubscriberLayer(this.state);
+    }
+  }
+
+  applySubscription() {
+    if (this.state.name.length && this.state.sliceId) {
+      const subscription = {};
+
+      Object.keys(this.state).forEach((k) => {
+        if (this.state[k] !== null) {
+          subscription[k] = this.state[k];
+        }
+      });
+
+      subscription['actions'] = [
+        "APPLY_FILTER"
+      ];
+
+      subscription['linked_slice'] = [
+        {
+          publisher_id: this.state.sliceId,
+          subscribe_columns: this.state.subscribe_columns,
+        }
+      ];
+
+      subscription['extras'] = this.state.extraValue;
+
+      this.props.addSubscriberLayer(subscription);
+      this.setState({ isNew: false, oldName: this.state.name });
+    }
+  }
+
+  submitSubscription() {
+    this.applySubscription();
+    this.props.close();
+  }
+
+  renderSingleSubscription(subscriptionData) {
+    const { columnType, operatorType, index } = subscriptionData;
+    const { allowColumnSelection, } = this.state;
+
+    const operators = this.getSupportedOperators();
+
+    return (
+      <div key={index} style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between', width: '400px' }}>
+
+        <SelectControl
+          hovered
+          description="Choose the Column"
+          disabled={!allowColumnSelection}
+          label="Select Column"
+          name="column-source-type"
+          options={this.state.publishedSliceColumns}
+          value={columnType}
+          onChange={(e) => this.handleColumnType(e, index)}
+        />
+
+        <SelectControl
+          hovered
+          description="Choose the Operator"
+          disabled={!allowColumnSelection}
+          label="Select Operator"
+          name="operator-source-type"
+          options={operators}
+          value={operatorType}
+          onChange={(e) => this.handleOperatorType(e, index)}
+        />
+
+        <Button title="Remove subscription columns and operators" bsSize="sm" disabled={index === 0} style={{ height: '30px', marginTop: '25px' }} onClick={(e) => this.removeColumn(e, subscriptionData)}>
+          {'-'}
+        </Button>
+
+      </div>
+    );
```

Comment:
can we move this as single jsx component and use it here

Reply:
we will do it in a separate US with validation tasks